### DISCUSSION TITLE
SendRequest Payload, i want to have multiple nodes based on prev request result

### DISCUSSION THREAD

**Discussion by gettosam** [postmanlabs/postman-flows#298] [Defect]
How do i construct the Request Body in the SEndRequest ,

i have something like this payload , i want to load the Childlist data into this payload from my previous data response, but my list may vary , for every different data i start testing with. 

as of now, my sendRequest has only 2 childlist i defined as below. how to make it more configurable childlist ?

{
field1: value,
field2: value,
childlist: {
[ 
   f1:{{f1var}},
   f2:{{f2var}}, 
   f3:{{f3var}}
],
[ 
   f1:{{f1var}},
   f2:{{f2var}}, 
   f3:{{f3var}}
]
}
}

**Comment by dev-kanishk** [comment]
Hey, One way to construct a Request body in the Send Request from the response of the previous block is to use, postman variables.

You can have the Payload which you mentioned above in the body of request using regular postman collection interface 

![Screenshot 2022-04-13 at 9 24 39 AM](https://user-images.githubusercontent.com/48204976/163097241-c6830e4e-4f51-4f46-a0e3-f3cf607ad051.png)

Then you can use Send Request block config for setting those variables.

![Screenshot 2022-04-13 at 9 29 00 AM](https://user-images.githubusercontent.com/48204976/163097507-c67809ee-8010-4e43-8d3c-1f68640170b7.png)

And the Flow looks like this

![Screenshot 2022-04-13 at 9 31 47 AM](https://user-images.githubusercontent.com/48204976/163097741-ce1a6e99-4008-4d57-b241-f8228c35137f.png)


Let me know if this helps you and any other query related to flows I will be happy to answer.

**Reply by gettosam** [reply]
sorry, this is straightforward variable replacing. i meant to ask, i have 5 set of child items, so then i want to have the SendRequest post body with 5 child nodes like  below. in your example, variables name comes always from index 0, we need to mention to retrieves all 5 names from get
childlist: {
[
f1:{{f1var}},
f2:{{f2var}},
f3:{{f3var}}
],
[
f1:{{f1var}},
f2:{{f2var}},
f3:{{f3var}}
]
}
